### PR TITLE
fix: handle parser error when couldn't parse

### LINKED ISSUE #39: Quick Tests
- Check that this works:
`{{ 'Hello {{ name }}' | translate: { name: 'Srinivas' } }}`
- Check that it works with `ngFor`:
```html
<component *ngFor="let a of array" headerText="{{ a | translate }}"></component>
```

### PR DESCRIPTION
**Pull request by itayod**
### added e2e test for loop

close #39

### CODE REVIEW

File: src/assets/i18n/en.json [hunk 1]
```diff
@@ -6,5 +6,7 @@
     "b": {
       "c": "a.b.c {{fromList}} english"
     }
-  }
+  },
+  "b": "b english",
```
Comment (NetanelBasal) [inline src/assets/i18n/en.json]: You need to add this in the typescript files too.

File: projects/transloco/src/lib/transloco.parser.ts [hunk 2]
```diff
@@ -5,11 +5,15 @@ import { getValue } from './helpers';
 export const TRANSLOCO_PARSER = new InjectionToken('TRANSLOCO_PARSER');
 
 export abstract class TranslocoParser {
-  abstract parse(value: string, params: HashMap, lang: HashMap): string;
+  abstract parse(value: string, params: HashMap, lang: HashMap, onError?: Function): string;
 }
 
 export class DefaultParser extends TranslocoParser {
-  parse(value: string, params: HashMap = {}, lang?: HashMap) {
+  parse(value: string, params: HashMap = {}, lang?: HashMap, onError?: Function): string {
```
Comment (NetanelBasal) [inline projects/transloco/src/lib/transloco.parser.ts]: I think it's redundant. We don't need to bloat the API. If someone really needs it, he can provide its own implementation.

File: projects/transloco/src/lib/transloco.parser.ts [hunk 3]
```diff
@@ -5,11 +5,15 @@ import { getValue } from './helpers';
...
+    if (typeof value !== 'string') {
+      return onError ? onError() && value : value;
+    }
```
Comment (NetanelBasal) [inline projects/transloco/src/lib/transloco.parser.ts]: You can change it to:
```ts
    return isString(value)
      ? value.replace(/{{(.*?)}}/g, function(_, match) {
          match = match.trim();
          return params[match] || getValue(lang, match) || '';
        })
      : value;
```

File: src/app/home/home.component.html [hunk 4]
```diff
@@ -40,3 +40,8 @@ <h3 class="mtb">Missing key</h3>
 <ul class="list-group">
   <li class="list-group-item" data-cy="missing-key"><b>Open the console </b>{{ 'alertty' | transloco }}</li>
 </ul>
+
+<h3 class="mtb">Translation loop</h3>
```
Comment (NetanelBasal) [inline src/app/home/home.component.html]:
```suggestion
<h3 class="mtb">Translation in ngFor</h3>
```

File: projects/transloco/src/lib/tests/transloco.parser.spec.ts [hunk 5]
```diff
@@ -34,4 +35,15 @@ describe('TranslocoParser', () => {
     expect(parser.parse(null)).toEqual(null);
     expect(parser.parse(undefined)).toEqual(undefined);
   });
+
+  it('shoudl call error handler when value can not be parsed', () => {
+    const errorCB = createSpy();
```
Comment (NetanelBasal) [inline projects/transloco/src/lib/tests/transloco.parser.spec.ts]: It's more common to do `jasmine.createSpy()`

**Comment by itayod** [comment]
@NetanelBasal updated the pr

File: projects/transloco/src/lib/tests/transloco.parser.spec.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import { DefaultParser } from '../../public-api';
+import createSpy = jasmine.createSpy;
```
Comment (NetanelBasal) [inline projects/transloco/src/lib/tests/transloco.parser.spec.ts]: remove

**Pull request merged**